pkg/prometheus/promcfg.go

```diff
@@ -2064,6 +2078,9 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	store assets.StoreGetter,
 	role string,
 	attachMetadataConfig *attachMetadataConfig,
+	selector *metav1.LabelSelector,
+	selectorMechanism *monitoringv1.SelectorMechanism,
+	selectorRoles []string,
```

Comment:
I'm not a fan of adding more parameters to this method. Maybe we need to refactor generateK8SSDConfig() so it does less? (the comment could also apply to the `attachMetadataConfig` parameter).
One option would be to introduce wrappers

```
appendRoleSelectors(
    attachMetaConfig.append(
        generateK8SSDConfig(...),
        ...,
    )
    ...,
)
````